common/sw-update/dev-updates-profile.cpp

```diff
@@ -38,47 +47,43 @@ namespace rs2
 
             bool update_available(false);
 
-            // We expect to get here in recovery mode (on firmware update flow) and therefore do not want to throw...
-            if (_update_profile.device_name.find("Recovery") == std::string::npos)
-            {
-                auto &versions_vec((comp == FIRMWARE) ?
-                    _update_profile.firmware_versions : _update_profile.software_versions);
+            auto &versions_vec((comp == FIRMWARE) ?
+                _update_profile.firmware_versions : _update_profile.software_versions);
 
-                version& current_version((comp == FIRMWARE) ? _update_profile.firmware_version : _update_profile.software_version);
+            version& current_version((comp == FIRMWARE) ? _update_profile.firmware_version : _update_profile.software_version);
```

Comment:
If it's not meant to be written to, I'd suggest a `const &` -- same for `versions_vec`